fabfile.py

```diff
@@ -13,13 +13,41 @@
 CURRENT_DIR = os.path.dirname(__file__)
 
 
+TIERS = {
+    'production': {
+        'hosts': ['root@104.131.92.59'],
+    },
+    'staging': {
+        'hosts': ['root@162.243.226.226'],
+    },
+}
+
+
+def tier_set(tier_name='staging'):
+    env.tier = tier_name
+    for option, value in TIERS[env.tier].items():
+        setattr(env, option, value)
+
+
+@task
+def production():
+    tier_set('production')
+
+
+@task
+def staging():
+    tier_set('staging')
+
+
 @task
 def install_salt():
+    require('tier', provided_by=(staging,production))
```

Comment:
It doesn't look to me like fabric has a built-in way to do that, but since "it's just python" we can do something like this, which doesn't seem too bad to me:

```python
@task
def foo():
    if env.tier == 'production':
        abort('cannot be run on production')
    run("uname -a")
```